src/proxy-ui-api/src/main/java/org/niis/xroad/restapi/service/NotificationService.java

```diff
@@ -0,0 +1,129 @@
+/**
+ * The MIT License
+ * Copyright (c) 2018 Estonian Information System Authority (RIA),
+ * Nordic Institute for Interoperability Solutions (NIIS), Population Register Centre (VRK)
+ * Copyright (c) 2015-2017 Estonian Information System Authority (RIA), Population Register Centre (VRK)
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.niis.xroad.restapi.service;
+
+import ee.ria.xroad.common.CodedException;
+import ee.ria.xroad.commonui.SignerProxy;
+import ee.ria.xroad.signer.protocol.dto.TokenInfo;
+
+import lombok.extern.slf4j.Slf4j;
+import org.niis.xroad.restapi.dto.AlertStatus;
+import org.niis.xroad.restapi.facade.GlobalConfFacade;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.time.OffsetDateTime;
+import java.time.ZoneOffset;
+import java.util.Optional;
+
+/**
+ * service class for handling notifications
+ */
+@Slf4j
+@Service
+@Transactional
+@PreAuthorize("isAuthenticated()")
+public class NotificationService {
+    private static OffsetDateTime backupRestoreRunningSince;
+    private final GlobalConfFacade globalConfFacade;
+    private final TokenService tokenService;
+
+    /**
+     * constructor
+     */
+    @Autowired
+    public NotificationService(GlobalConfFacade globalConfFacade, TokenService tokenService) {
+        this.globalConfFacade = globalConfFacade;
+        this.tokenService = tokenService;
+    }
+
+    /**
+     * Checks the status of system alerts that may affect whether the system
+     * is operational or not. If alerts are disabled, the status of all alerts true.
+     * @return
+     */
+    public AlertStatus getAlerts() {
```

Comment:
Should this also check whether the Security Server has been initialized yet? That is of course still in review but since the old implementation checks the init status, maybe it should be added here also to avoid throwing "false" exceptions. If it is something that is wanted, maybe we should create a new ticket so it does not get forgotten.

Reply:
Yes, you're right. It should be checked whether the Security Server is initialized or not. I'll create a new ticket about it that can be implemented once XRDDEV-904 has been completed.

Reply:
The init status check will be implemented in XRDDEV-1035 (https://jira.niis.org/browse/XRDDEV-1035).